jenkins_pipelines/scripts/maintenance_json_generator.py

```diff
@@ -2,139 +2,143 @@
 import json
 import subprocess
 import sys
-
 import requests
+import xml.etree.ElementTree as ET
+from datetime import date, datetime
+from functools import cache
+
+CURRENT_DATE: date = date.today()
 
 # dictionary for 4.3 client tools
-clienttools43dict = {
-    "sle12sp5_client": "/SUSE_Updates_SLE-Manager-Tools_12_x86_64/",
-    "sle12sp5_minion": "/SUSE_Updates_SLE-Manager-Tools_12_x86_64/",
-    "sle15_client": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
-                     "/SUSE_Updates_SLE-Product-SLES_15-LTSS_x86_64/"],
-    "sle15_minion": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
-                     "/SUSE_Updates_SLE-Product-SLES_15-LTSS_x86_64/"],
-    "sle15sp1_client": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
-                        "/SUSE_Updates_SLE-Product-SLES_15-SP1-LTSS_x86_64/"],
-    "sle15sp1_minion": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
-                        "/SUSE_Updates_SLE-Product-SLES_15-SP1-LTSS_x86_64/"],
-    "sle15sp2_client": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
-                        "/SUSE_Updates_SLE-Product-SLES_15-SP2-LTSS_x86_64/"],
-    "sle15sp2_minion": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
-                        "/SUSE_Updates_SLE-Product-SLES_15-SP2-LTSS_x86_64/"],
-    "sle15sp3_client": ["/SUSE_Updates_SLE-Manager-Tools_15_x86_64/",
+clienttools43dict: dict[str, set[str]] = {
```

Comment:
Good practice is to do not express the type inside the variable name e.g. `my_list: list`, so I would just go for:
```suggestion
clienttools43: dict[str, set[str]] = {
```

The same with other variable names covered by this PR.

Reply:
Right, I kept the previous names but they do not make make sense once type hints are in place.